corokafka/corokafka_connector.h

```diff
@@ -35,7 +31,7 @@ class ConnectorImpl;
  *        setup the required topics for publishing/consumption and to get the ProducerManger and/or
  *        ConsumerManager objects for interacting with these topics.
  */
-class Connector
+class Connector : public Impl<IConnector>
```

Comment:
I'll have more time for this later, but an initial thought is that we might want to ease use by renaming Connector to something like ConnectorBase, and adding 
using Connector = ConnectorBase<Impl>; so that by default, the developer just continues to use Connector and gets the real thing.